apis/datadoghq/v2alpha1/datadogagent_types.go

```diff
@@ -6,28 +6,209 @@
+	Name string `json:"name,omitempty"`
+}
+
+// DatadogAgentPodTemplateOverride is the generic description equivalent to a subset of the PodTemplate for a component
+type DatadogAgentPodTemplateOverride struct {
+	// Agent config contain the basic configuration of the Datadog Process Agent's container.
+	Containers []DatadogAgentGenericContainer `json:"containers,omitempty"`
+
+	// Specify additional volumes in the different components (Datadog Agent, Cluster Agent, Cluster Check Runner).
+	// +optional
+	// +listType=map
+	// +listMapKey=name
+	Volumes []corev1.Volume `json:"volumes,omitempty"`
+
+	// The container image of the different components (Datadog Agent, Cluster Agent, Cluster Check Runner).
+	Image *ImageConfig `json:"image,omitempty"`
+
+	// Configure the different components (Datadog Agent, Cluster Agent, Cluster Check Runner) tolerations.
+	// +optional
+	// +listType=atomic
+	Tolerations []corev1.Toleration `json:"tolerations,omitempty"`
+
+	// Pod-level SecurityContext.
+	// +optional
+	SecurityContext *corev1.PodSecurityContext `json:"securityContext,omitempty"`
+
+	// If specified, indicates the pod's priority. "system-node-critical" and "system-cluster-critical"
+	// are two special keywords which indicate the highest priorities with the former being the highest priority.
+	// Any other name must be defined by creating a PriorityClass object with that name. If not specified,
+	// the pod priority will be default or zero if there is no default.
+	PriorityClassName string `json:"priorityClassName,omitempty"`
+
+	// If specified, the pod's scheduling constraints.
+	// +optional
+	Affinity *corev1.Affinity `json:"affinity,omitempty"`
+
+	// Annotations provide annotations that will be added to the different components (Datadog Agent, Cluster Agent, Cluster Check Runner) Pods.
+	Annotations map[string]string `json:"annotations,omitempty"`
+
+	// AdditionalLabels provide labels that will be added to the different components (Datadog Agent, Cluster Agent, Cluster Check Runner) Pods.
+	Labels map[string]string `json:"labels,omitempty"`
+}
+
+// DatadogAgentGenericContainer is the generic structure describing any container's common configuration.
+// +k8s:openapi-gen=true
+type DatadogAgentGenericContainer struct {
+
+	// Name of the container that is overiden
+	//+optional
+	Name string `json:"name,omitempty"`
+
+	// The different components (Datadog Agent, Cluster Agent, Cluster Check Runner) support many environment variables.
+	// See also: https://docs.datadoghq.com/agent/kubernetes/?tab=helm#environment-variables
+	//
+	// +optional
+	// +listType=map
+	// +listMapKey=name
+	Env []corev1.EnvVar `json:"env,omitempty"`
+
+	// Specify additional volume mounts in the container.
+	// +optional
+	// +listType=map
+	// +listMapKey=name
+	// +listMapKey=mountPath
+	VolumeMounts []corev1.VolumeMount `json:"volumeMounts,omitempty"`
+
+	// Make sure to keep requests and limits equal to keep the pods in the Guaranteed QoS class.
```

Comment:
should we write something more generic here? (or, consider having this on a second line and write something generic above it)